src/app/goals/goals-list/goals-list.component.ts

```diff
@@ -42,6 +46,19 @@ export class GoalsListComponent implements OnInit, OnDestroy {
       this.goals = goals;
       this.selectTheFirstGoal();
     });
+
+    this.filter.valueChanges
+      .pipe(
+        distinctUntilChanged(),
+        debounceTime(200),
```

Comment:
Do we need debounceTime? It's not a request to server, I think immediate response would be better here.

Reply:
BTW nice avatar photo maaaan

Reply:
Thanks :)